internal/uidriver/glfw/ui.go

```diff
@@ -487,27 +487,39 @@ func (u *UserInterface) adjustPosition(x, y int) (int, int) {
 	return x - int(ox/s), y - int(oy/s)
 }
 
-func (u *UserInterface) IsCursorVisible() bool {
+func (u *UserInterface) CursorMode() driver.CursorMode {
 	if !u.isRunning() {
-		return u.isInitCursorVisible()
+		return u.getInitCursorMode()
 	}
-	v := false
+	v := driver.CursorModeVisible
 	_ = u.t.Call(func() error {
-		v = u.window.GetInputMode(glfw.CursorMode) == glfw.CursorNormal
+		switch u.window.GetInputMode(glfw.CursorMode) {
+		case glfw.CursorHidden:
+			v = driver.CursorModeHidden
+		case glfw.CursorDisabled:
+			v = driver.CursorModeCaptured
+		}
 		return nil
 	})
 	return v
 }
 
-func (u *UserInterface) SetCursorVisible(visible bool) {
+func (u *UserInterface) SetCursorMode(mode driver.CursorMode) {
 	if !u.isRunning() {
-		u.setInitCursorVisible(visible)
+		u.setInitCursorMode(mode)
 		return
 	}
 	_ = u.t.Call(func() error {
-		c := glfw.CursorNormal
-		if !visible {
+		c := glfw.NoAPI
```

Comment:
`NoAPI` as a default value does not make sense. Let's panic when `mode` is an invalid value.

Reply:
I have added a panic here. However, I have left the initial assignment of c := glfw.NoAPI. This is so I can guarantee the resolution of const to the correct type. If you have other suggestions, I can make changes.

Reply:
`var c int` would make sense as `SetInputMode` takes an integer value.